File: common-boot-starter/src/main/java/com/dasoops/common/entity/param/base/IBuildDo.java
package com.dasoops.common.entity.param.base;

import cn.hutool.core.bean.BeanUtil;
import com.dasoops.common.entity.dbo.base.BaseDo;
import com.dasoops.common.util.ReflectUtil;

/**
 * @Title: IBuildDo
 * @ClassPath com.dasoops.common.entity.param.base.IBuildDo
 * @Author DasoopsNicole@Gmail.com
 * @Date 2023/01/13
 * @Version 1.0.0
 * @Description: 构建do
 */
public interface IBuildDo<T extends BaseDo> {

    /**
     * 构建do
     *
     * @return {@link T}
     */
    default T buildDo() {
        T t = ReflectUtil.getGenericInstance(getClass(), 0);
        BeanUtil.copyProperties(this, t);
        return t;
    }

}

File: common-boot-starter/src/main/java/com/dasoops/common/entity/param/base/IBuildDo.kt
package com.dasoops.common.entity.param.base

import cn.hutool.core.bean.BeanUtil
import com.dasoops.common.entity.dbo.base.BaseDo
import com.dasoops.common.util.ReflectUtil

/**
 * @Title: IBuildDo
 * @ClassPath com.dasoops.common.entity.param.base.IBuildDo
 * @Author DasoopsNicole@Gmail.com
 * @Date 2023/01/13
 * @Version 1.0.0
 * @Description: 构建do
 */
interface IBuildDo<T : BaseDo> {
    /**
     * 构建do
     *
     * @return [T]
     */
    fun buildDo(): T {
        val t = ReflectUtil.getGenericInstance<T>(javaClass, 0)
        BeanUtil.copyProperties(this, t)
        return t
    }
}